### PR TITLE
Fix(transactions): Do not return completed transactions

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?
Not awaited async functions that used transactions had a weird behaviour.

If you tried , in a straipi bootsrap:
```js
const insertTag = async (strapi) => {
  await strapi.entityService.create('api::tag.tag', {
    data: {
      name: 'tag4',
    },
  });
};

    await strapi.db.transaction(async (trx) => {
      // Without awaiting
      insertTag(strapi);
    });
```

Would result in the following error:
```
This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
Error: Transaction query already complete, run with DEBUG=knex:tx for more info
```

TLDR: Any async code running inside a transaction	will throw an error.

### Why is it needed?

This edge case happens for any event like subscriber there is in strapi (audit logs, event hub, ...)
During the development of Review Workflows, we noticed an audit logs error like this one, and it was only caused when we were using `strapi.db.transactions`

Everything inside the strpai.db.transaction should be encapsulated inside the same transaction, but if there is any async code execution and the transaction is already completed at the time it is executed, we should provide an alternative.

### How to test it?

In `getstarted/src/index.js`
```js
'use strict';

const insertTag = async (strapi) => {
  await strapi.entityService.create('api::tag.tag', {
    data: {
      name: 'tag1',
    },
  });
};
module.exports = {
  async bootstrap({ strapi }) {
    await strapi.db.transaction(async (trx) => {
      insertTag(strapi);
    });
  },

};

```

This should not print any error.

### CODE REVIEW

File: packages/core/database/lib/transaction-context.js [hunk 1]
```diff
@@ -10,7 +10,8 @@ const transactionCtx = {
   },
 
   get() {
-    return storage.getStore();
+    const trx = storage.getStore();
+    return trx?.isCompleted() ? undefined : trx;
```
Comment (nathan-pichon) [inline packages/core/database/lib/transaction-context.js:14]: Should we automatically remove the local stored value when transactions are done?

File: packages/core/database/lib/index.js [hunk 2]
```diff
@@ -59,13 +59,16 @@ class Database {
       if (notNestedTransaction) {
         await trx.commit();
       }
+      transactionCtx.clear();
```
Comment (Bassel17) [inline packages/core/database/lib/index.js:62]: I would suggest to move this to the if statement, otherwise it will break the nested transactions and the context will be cleared when it shouldn't

**Review by Bassel17 — CHANGES_REQUESTED** [review]

**Review by Bassel17 — APPROVED** [review]
LGTM
